### PR TITLE
A few documentation improvements

### PR DESCRIPTION
**Pull request by Namoshek**
I started going through the documentation of OCPI 2.2-rc1 and fixed some of the spelling/wording issues I found. I also highlighted some occurences of `should`, `may` or `must` by writing them entirely uppercase where I found them to be important.

Before going further with this, are such contributions actually desired?
I realize that not all of my changes are _very minor spelling/wording alternations_. For example, in `credentials.asciidoc` I added `ISO-3166 alpha-2` as source for country codes as it seems the only logical source to me. One could see this as unwanted constraint, I see it as suggestion of good practice. Therefore I'd appreciate some feedback before going on with what I'm doing so far.

### CODE REVIEW

**Comment by poohsen** [comment]
@RobertdeLeeuw-IHomer what do you think about the capitalized verbs?

**Comment by RobertdeLeeuw-IHomer** [comment]
I did not have time to look at this, first will work on OCPI 2.1.1-d2 now. 
I already have done some of the words in capitals. I normaly tend to like it, but we then also should add a reference to: https://www.ietf.org/rfc/rfc2119.txt. 
but also need to be very consistent with it. Which we have not yet been. But if @Namoshek has done all of that, I'm in favor of doing it.

**Comment by Namoshek** [comment]
As of now, I haven't been a 100% consistent with capitalized verbs as I'm not convinced that every occurance of a verb is equally important. But we should at least capitalize the important usages of DOs and DONTs (where ambiguity is given if implementations do what they want). The usages of _should_ are not too important in my opinion but a nice to have.

File: transport_and_format.asciidoc [hunk 1]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 
 For simplicity, connected clients might PUSH (POST, PUT, PATCH) information to all connected clients with an "opposite role",
 CPO pushing information to all eMSPs and NSPs. eMSP pushing information to all CPOs.
-(the role "Other" is seen as an eMSP type of role, so broadcast push from a CPO is also send to Other,
-messages from "Other" are send to CPOs, not the eMSPs.)
+(The role "Other" is seen as an eMSP type of role, so broadcast push from a CPO is also sent to "Other".
+Messages from "Other" are only sent to CPOs and not to eMSPs though.)
 
-Broadcast push might be very useful to push information like Locations or Tokens to all parties,
+Broadcast pushing data through a Hub might be very useful to share information like Locations or Tokens with all parties
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: I would like to use 'Broadcast Push' as a term for a functionality. Maybe naming it Broadcast Push here would have made it more clear.

File: transport_and_format.asciidoc [hunk 2]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
-This means only 1 request to the Hub, not having to worry about the number of connected clients.
+This means only one request to the Hub will be necessary, as all connected clients will be served by the Hub.
+In other words: the Hub broadcasts received information to all connected clients.
 
-To send a Broadcast push, the client uses to Hubs party-id and country-code in the 'OCPI-to-' headers.
-The Hub parses the request and sends a response to the client, with optionally its own party-id and country-code in the 'OCPI-from-' headers.
-The Hub then sends the push to any client implementing the applicable module, using its own party-id and country-code in the 'OCPI-from-' headers.
-The client receiving a push from a Hub (with the Hubs information in the 'OCPI-from-' headers) will respond to
-this push with the Hubs party-id and country-code in the 'OCPI-to-' headers.
+To send a broadcast push, the client uses the party-id and country-code of the Hub in the 'OCPI-to-' header.
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: Again, maybe: Broadcast Push is more clear to explain this is a term for a certain functionality

File: transport_and_format.asciidoc [hunk 3]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
+The Hub parses the request and sends a response to the client, which optionally contains its own party-id and country-code in the 'OCPI-from-' header.
+The Hub then sends the pushed data to any client implementing the corresponding applicable module, using its own party-id and country-code in the 'OCPI-from-' header.
+The client receiving a PUSH from a Hub (with the Hubs information in the 'OCPI-from-' header) will respond to
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: Why do you choose to use all caps: PUSH here?

File: transport_and_format.asciidoc [hunk 4]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
+this PUSH with the Hubs party-id and country-code in the 'OCPI-to-' header.
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remark

File: transport_and_format.asciidoc [hunk 5]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 
-GET SHALL NOT be used in combination with Broadcast push.
+GET SHALL NOT be used in combination with broadcast pushing.
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remark

File: transport_and_format.asciidoc [hunk 6]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 If the requesting party wants to GET information of which it does not know the receiving party,
-to <<transport_and_format_message_routing_open_routing_request,Open routing request>> MUST be used. (see below)
+an <<transport_and_format_message_routing_open_routing_request,open routing request>> MUST be used. (see below)
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: As with `Broadcast Push`, I want to introduce the term: `Open Routing Request`, so maybe would have been better if I had used it this way. But I'm open for better names.

File: transport_and_format.asciidoc [hunk 7]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 
-Broadcast push SHALL only be used with information that is meant to be send to all other parties. It is useful for things like:
+Broadcast pushing SHALL only be used with information that is meant to be sent to all other parties. It is useful to share data like
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remark

File: transport_and_format.asciidoc [hunk 8]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 <<mod_tokens.asciidoc#mod_tokens_tokens_module,Tokens>> and <<mod_locations.asciidoc#mod_locations_locations_module,Locations>>,
-but not for <<mod_cdrs.asciidoc#mod_cdrs_cdrs_module,CDRs>> and <<mod_sessions.asciidoc#mod_sessions_sessions_module,Sessions>>.
+but not so much for <<mod_cdrs.asciidoc#mod_cdrs_cdrs_module,CDRs>> and <<mod_sessions.asciidoc#mod_sessions_sessions_module,Sessions>>
+as these pieces of information are specific to only one party and are possibly even protected by GDPR or other laws.
 
-NOTE: For "Client owned objects" the party-id and country-code in the URL segments will still be the original party-id and country-code from the original client sending the Broadcast push to the Hub.
+NOTE: For "client owned objects", the party-id and country-code in the URL segments will still be the original party-id and country-code from the original client sending the Broadcast push to the Hub.
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: Also `Client Owned Objects` is a sort of term used in OCPI. So maybe starting with capitals is better.

File: transport_and_format.asciidoc [hunk 9]
```diff
@@ -312,81 +312,83 @@ include::examples/transport_and_format_get_token_list_example.json[]
...
 image::images/sd_put_boardcast.svg[OCPI Sequence Diagram Hub PUT]
 
 
 [[transport_and_format_message_routing_open_routing_request]]
 ===== Open routing request
-When a Hub has the intelligence to route messages, based on the content of the request,
-or the requesting party does not know the destination of a request, the 'OCPI-to-' headers can be omitted in the request towards a hub.
-The Hub can then decide to which part a request needs to be routed, or that it needs to be broadcasted.
+When a Hub has the intelligence to route messages based on the content of the request,
+or the requesting party does not know the destination of a request, the 'OCPI-to-' header can be omitted in the request towards the Hub.
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: is `header` correct? there are 2 headers: `OCPI-to-party-id` and `OCPI-to-country-code`

File: transport_and_format.asciidoc [hunk 10]
```diff
@@ -445,22 +447,22 @@ This table contains the description of which headers are required to be used for
...
 | Name | Route | TO Header | FROM Header | Description
-| Direct request | Requesting party to Hub | Receiving-party | | FROM headers are optional, Hub knows the requesting party
-| Direct request | Hub to receiving party | | Requesting-party | TO headers are optional, party knows this request is for the party itself.
-| Direct response | receiving party to Hub | Requesting-party | | FROM headers are optional, Hub knows the responding party
-| Direct response | Hub to requesting party | | Receiving-party | TO headers are optional, party knows this response is for the party itself.
+| Direct request | Requesting party to Hub | Receiving-party | | FROM header is optional, Hub knows the requesting party.
+| Direct request | Hub to receiving party | | Requesting-party | TO header is optional, party knows this request is for the party itself.
+| Direct response | Receiving party to Hub | Requesting-party | | FROM header is optional, Hub knows the responding party.
+| Direct response | Hub to requesting party | | Receiving-party | TO header is optional, party knows this response is for the party itself.
 |===
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remarks, it is always a set of 2 headers

File: transport_and_format.asciidoc [hunk 11]
```diff
@@ -470,42 +472,42 @@ image::images/sd_virtual_hub_virtual.svg[Example sequence diagram of a GET for 1
...
 | Name | Route | TO Header | FROM Header | Description
-| Broadcast request | Requesting party to Hub | Hub | | FROM headers are optional, Hub knows the requesting party
-| Broadcast response | Hub to requesting party | | Hub | TO headers are optional, party knows this response is for the party itself.
-| Broadcast request | Hub to receiving party | | Hub | TO headers are optional, party knows this request is for the party itself.
-| Broadcast response | receiving party to Hub | Hub | | FROM headers are optional, Hub knows the responding party
+| Broadcast request | Requesting party to Hub | Hub | | FROM header is optional, Hub knows the requesting party.
+| Broadcast response | Hub to requesting party | | Hub | TO header is optional, party knows this response is for the party itself.
+| Broadcast request | Hub to receiving party | | Hub | TO header is optional, party knows this request is for the party itself.
+| Broadcast response | Receiving party to Hub | Hub | | FROM header is optional, Hub knows the responding party.
 |===
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remarks, it is always a set of 2 headers

File: transport_and_format.asciidoc [hunk 12]
```diff
@@ -470,42 +472,42 @@ image::images/sd_virtual_hub_virtual.svg[Example sequence diagram of a GET for 1
...
-| Open request | Requesting party to Hub | | | Omitting the TO headers indicates to the Hub that the Hub has to figure out the routing.
-| Open request | Hub to receiving party | | Requesting-party | TO headers can be omitted, when the receiving party is NOT a virtual party.
-| Open response | receiving party to Hub | Requesting-party | Receiving-party | FROM headers can be omitted, when the receiving party is NOT a virtual party.
-| Open response | Hub to requesting party | Requesting-party | Receiving-party | TO headers can be omitted, when the requesting party is NOT a virtual party.
+| Open request | Requesting party to Hub | | | Omitting the TO header indicates to the Hub that the Hub has to figure out the routing itself.
+| Open request | Hub to receiving party | | Requesting-party | TO header can be omitted when the receiving party is NOT a virtual party.
+| Open response | Receiving party to Hub | Requesting-party | Receiving-party | FROM header can be omitted when the receiving party is NOT a virtual party.
+| Open response | Hub to requesting party | Requesting-party | Receiving-party | TO header can be omitted when the requesting party is NOT a virtual party.
 |===
```
Comment (RobertdeLeeuw-IHomer) [inline transport_and_format.asciidoc]: See previous remarks

File: version_information_endpoint.asciidoc [hunk 13]
```diff
@@ -176,27 +175,27 @@ List of known versions.
...
+It is advised to use a prefix (e.g. country-code + party-id) for any custom moduleID, this ensures that the moduleID will not be used for any future module of OCPI.
 
 For example:
 `nltnm-tokens`
 
 [[version_information_get_endpoint]]
 ==== GET
 
-Fetch information about the supported endpoints and their URLs for this version.
+Fetch information about the supported endpoints and their URLs for this OICP version.
```
Comment (RobertdeLeeuw-IHomer) [inline version_information_endpoint.asciidoc]: OICP? No, that is a different protocol: OCPI

File: mod_locations.asciidoc [hunk 14]
```diff
@@ -6,43 +6,44 @@
 
 *Data owner: `CPO`*
 
-The Location objects live in the CPO back-end system. They describe the charging locations of that operator.
+The Location objects live in the CPO back-end system. They describe the charging locations of one operator (or of multiple operators in case of virtual CPOs).
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: I think we will remove the term: Virtual CPO. The ideas it to describe `OCPI Platforms` and then each platform can provide services for different CPO's. So I don't think 1 CPO will report a Location that belongs to multiple CPO's. Then all the CPO's will have to report a Location with the same information GPS coordinates etc, but different EVSE's.

File: mod_locations.asciidoc [hunk 15]
```diff
@@ -6,43 +6,44 @@
...
+The Locations module has Location as base object. Locations can have multiple EVSEs and EVSEs can have multiple Connectors.
 With the methods in the <<mod_locations_emsp_interface,eMSP interface>>,
 Location information/statuses can be shared with the eMSP.
-Updates can be done to the Location, but also to only an EVSE or a Connector.
+Updates can be made to a whole Location, but also to only an EVSE or a Connector.
 
-When a CPO creates Location objects it pushes them to the eMSPs by calling <<mod_locations_put_method,PUT>> on the eMSPs Locations endpoint.
-Providers who do not support push mode need to call <<mod_locations_get_method,GET>> on the CPOs Locations endpoint to receive the new object.
+When a CPO creates Location objects, it pushes them to the eMSPs by calling <<mod_locations_put_method,PUT>> on the eMSPs Locations endpoint.
+Providers who do not support PUSH mode need to call <<mod_locations_get_method,GET>> on the CPOs Locations endpoint to receive the new object.
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: Would all caps `PUSH` be more clear? Then I/we need to check the entire document for this. And then also for `PULL`.

File: mod_locations.asciidoc [hunk 16]
```diff
@@ -6,43 +6,44 @@
...
 
 If the CPO wants to replace a Location related object,
 they push it to the eMSP systems by calling <<mod_locations_put_method,PUT>> on their Locations endpoint.
 
 Any changes to a Location related object can also be pushed to the eMSP by calling the <<mod_locations_patch_method,PATCH>>
-on the eMSPs Locations endpoint. Providers who do not support push mode need to
+on the eMSPs Locations endpoint. Providers who do not support PUSH mode need to
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: See previous remarks

File: mod_locations.asciidoc [hunk 17]
```diff
@@ -6,43 +6,44 @@
...
 
 [[mod_locations_interfaces_and_endpoints]]
 === Interfaces and endpoints
 
-There is both a CPO and an eMSP interface for Locations. Advised is to use the push direction from CPO to eMSP during normal operation.
-The CPO interface is meant to be used when the connection between 2 parties is established, to retrieve the current list of Location objects with the current status, and when the eMSP is not 100% sure the Locations cache is completely correct.
-The eMSP can use the CPO GET Object interface to retrieve a specific Location, EVSE or Connector, this might be used by a eMSP that wants information about a specific Location, but has not implemented the eMSP Locations interface (cannot receive push).
+There is both a CPO and an eMSP interface for Locations. It is advised to use the PUSH direction from CPO to eMSP during normal operation.
+The CPO interface is meant to be used when the connection between two parties is established, to retrieve the current list of Location objects with the current status, and when the eMSP is not 100% sure the Location cache is entirely up-to-date.
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: See previous remarks `PUSH`

File: mod_locations.asciidoc [hunk 18]
```diff
@@ -6,43 +6,44 @@
...
+The eMSP can use the CPO GET Object interface to retrieve a specific Location, EVSE or Connector.
+This feature might be used by an eMSP that wants information about a specific Location, but has not implemented the eMSP Locations interface (i.e. cannot receive PUSH).
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: See previous remarks

File: mod_locations.asciidoc [hunk 19]
```diff
@@ -83,8 +84,8 @@ Examples:
 `+https://www.server.com/ocpi/cpo/2.2/locations/?offset=50&amp;limit=100+`
 
 
-If additional parameters: {date_from} and/or {date_to} are provided, only Locations with (`last_updated`) between the given date_from and date_to will be returned.
-If an EVSE is updated, also the 'parent' Location's `last_updated` fields is updated. If a Connector is updated, the EVSE's `last_updated` and the Location's `last_updated` field are updated.
+If additional parameters `{date_from}` and/or `{date_to}` are provided, only Locations with (`last_updated`) between the given `{date_from}` and `{date_to}` will be returned.
+If an EVSE is updated, also the 'parent' Location's `last_updated` field needs to be updated. Analogously, if a Connector is updated, the EVSE's `last_updated` and the Location's `last_updated` fields need to be updated updated.
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: I don't like to use words that are bit more complex for most non native English speakers. Could we rephrase to not use: Analogously?

File: mod_locations.asciidoc [hunk 20]
```diff
@@ -13,36 +13,42 @@ The Location objects live in the CPO back-end system. They describe the charging
...
 If the CPO wants to replace a Location related object,
-they push it to the eMSP systems by calling <<mod_locations_put_method,PUT>> on their Locations endpoint.
+they again push it to the eMSP systems by calling <<mod_locations_put_method,PUT>> on their Locations endpoint.
 
-Any changes to a Location related object can also be pushed to the eMSP by calling the <<mod_locations_patch_method,PATCH>>
-on the eMSPs Locations endpoint. Providers who do not support PUSH mode need to
-call <<mod_locations_get_method,GET>> on the CPOs Locations endpoint to receive the updates.
+Any changes to a Location related object can also be pushed to connected eMSPs by calling the <<mod_locations_patch_method,PATCH>> method
+on the eMSPs Locations endpoint, but using PATCH mode, only actual changes should be pushed.
+Providers who do not support PUSH mode need to call <<mod_locations_get_method,GET>> on the CPOs Locations endpoint to receive the updates.
```
Comment (RobertdeLeeuw-IHomer) [inline mod_locations.asciidoc]: See previous remark: PUSH

File: transport_and_format.asciidoc (hunk 1 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: Yes, it makes it more clear. Personally, I think the term _broadcast_ already implies that data is _pushed_ and not _pulled_, but if you want to keep the term, that's fine for me. I'll change the occurences accordingly.

File: transport_and_format.asciidoc (hunk 3 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: I changed a few more occurences of it which are not shown in the diffs by GitHub for whatever reason:
> eMSPs who do not support PUSH mode need to call GET on the CPOs Locations endpoint to receive the new object.

> It is advised to use the PUSH direction from CPO to eMSP during normal operation [...].

> This feature might be used by an eMSP that wants information about a specific Location, but has not implemented the eMSP Locations interface (i.e. cannot receive PUSH). 

To me it simply feels right to _somehow_ pronounce that we are talking about PUSH/PULL. We implicitely do the same when we write `GET`, `POST`, `PATCH`, ... so why not in this case?

Other examples are _Broadcast Push_ and _Open Routing Request_ where we pronounce a specific term by writing them explicitely in a different way. In above examples, _Push Mode_ could be used as such term. But actually it would be rather easy to avoid the word `push` entirely:

> eMSPs who do not support ~~PUSH mode~~ **receiving data** need to call GET on the CPOs Locations endpoint to receive the new object.

> It is advised to use the ~~PUSH~~ direction from CPO to eMSP during normal operation [...].

> This feature might be used by an eMSP that wants information about a specific Location, but has not implemented the eMSP Locations interface (i.e. cannot receive ~~PUSH~~ **data**). 

However, the second example would be somewhat arbitrary.

File: transport_and_format.asciidoc (hunk 9 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: No, it is wrong and you are absolutely right. I'll change it. I'll also update the table headers in chapter 3.1.7.4 which currently use `TO Header` and `FROM Header` in singular form.

File: version_information_endpoint.asciidoc (hunk 13 above)
Comment (Namoshek) [reply version_information_endpoint.asciidoc]: How embarrassing. 😅

File: mod_locations.asciidoc (hunk 14 above)
Comment (Namoshek) [reply mod_locations.asciidoc]: I see how my addition can be read the way you construe it. As this is the initial description of the module, my intention was to cover all use-cases of the module. I didn't mean (and even considered) that data of multiple (virtual) CPOs could be mixed within the module or the protocol. To me this also doesn't make sense from a business perspective where I would find it very weird if company A is providing white-label services to company B and both companies A&B had a charging station at the very same location (or even if A provided white-label services to B&C and both B&C had a station at the same location).

I'll rephrase it to something more clear for now.

File: mod_locations.asciidoc (hunk 15 above)
Comment (Namoshek) [reply mod_locations.asciidoc]: In my opinion it would be more clear as written in another comment above in more detail.

File: mod_locations.asciidoc (hunk 19 above)
Comment (Namoshek) [reply mod_locations.asciidoc]: I hope _similarly_ is a more common word?

File: transport_and_format.asciidoc (hunk 8 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: What about the chapter heading `3.1.5. Client owned object push`? The `push` also capitalized?

File: transport_and_format.asciidoc (hunk 6 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: I capitalized the term, it had very few occurences anyway. See 1c8f2be72e5033e84f3ef5927f486d4c7eeeedec for a diff.

Personally, I find the term quite fitting, although I definitely had to read the documentation to understand the intention behind it.

**Comment by Namoshek** [comment]
https://github.com/ocpi/ocpi/blame/1a2c64d53c4c9e48543b947ef0542eec2199b695/mod_sessions.asciidoc#L157

What exactly is the intention behind the description of the `Set Charging Preferences` method? Until this point in the documentation I expected that one changes the charging preferences **for a single session**, but this reads like **it changes the charging preferences for the driver of the session (i.e. also for future sessions of the same driver)**.

**Comment by Namoshek** [comment]
What is the difference between _objects_ and _Classes_ (see chapters 7.3 and 7.4 as example)? Why are some types in their own chapter and called _objects_ while others that are documented the same way (equal table with the same headings) are called _classes_?

File: transport_and_format.asciidoc (hunk 3 above)
Comment (RobertdeLeeuw-IHomer) [reply transport_and_format.asciidoc]: GET, POST, PATCH etc are HTTP Methods that are always in all caps. 

Push and Pull are not very familiar terms for implementers of OCPI, removing the terms would not help them.

File: mod_locations.asciidoc (hunk 19 above)
Comment (RobertdeLeeuw-IHomer) [reply mod_locations.asciidoc]: similarly is good

**Comment by RobertdeLeeuw-IHomer** [comment]
> https://github.com/ocpi/ocpi/blame/1a2c64d53c4c9e48543b947ef0542eec2199b695/mod_sessions.asciidoc#L157
> 
> What exactly is the intention behind the description of the `Set Charging Preferences` method? Until this point in the documentation I expected that one changes the charging preferences **for a single session**, but this reads like **it changes the charging preferences for the driver of the session (i.e. also for future sessions of the same driver)**.

I see what you mean, wording is not so good. This needs to be something like: "Updated Charging preferences for this driver for this Session only"

**Comment by RobertdeLeeuw-IHomer** [comment]
> What is the difference between _objects_ and _Classes_ (see chapters 7.3 and 7.4 as example)? Why are some types in their own chapter and called _objects_ while others that are documented the same way (equal table with the same headings) are called _classes_?

This was already when I started working on the OCPI documentation. 

The idea? The Objects are the type of classes you can Push/Pull via this module. The classes below are sub classes of these classes.

**Comment by RobertdeLeeuw-IHomer** [comment]
BTW, I already started to extend the terminology list to include things like: Push, Pull, Broadcast Push etc.

**Comment by Namoshek** [comment]
> I see what you mean, wording is not so good. This needs to be something like: "Updated Charging preferences for this driver for this Session only"

I changed the text in 0a1156222fe61f4252c966c85c23ad8f908669a7.

> This was already when I started working on the OCPI documentation.
> 
> The idea? The Objects are the type of classes you can Push/Pull via this module. The classes below are sub classes of these classes.

I already expected this reasoning. But then there has been the response of the PUT method described in chapter 8.2.1.2. The response consists of a simple enum described in 8.4.1. which is situated under 8.4. Data Types. So it's not all of the types that are _transferred_ (my words)...

Anyway, I'd leave this point for another day.

> BTW, I already started to extend the terminology list to include things like: Push, Pull, Broadcast Push etc. 2c1ca56

Looks good to me. But a question: Do you also consider capitalized names to have either of these two meanings?
1. The term is OCPI specific like `Client Owned Object`.
2. The term has two meanings and we therefore differentiate by capitalizing it in certain situations. The lower case form describes something in _the real world_ (i.e. `session` as in _charging session_) while the capitalized form describes something heavily related to OCPI (i.e. `Session` as in _Session object_).

The second situation leads to following naming conventions in my opinion:
- `driver` as there is no `Driver` object/class
- `session` where the charging session in the physical world is meant
- `Session` where the OCPI object is meant
- `charging preferences` where the actual preferences/demands of a driver are described
- `ChargingPreferences` or `Charging Preferences` where the object/class or transferred data is meant
- `pull` as verb
- `Pull` where the action of pulling data is meant
- ...

Only issue with this is the ambiguity when we have to write words capitalized at the beginning of a sentence. Any suggestion for this case?

File: transport_and_format.asciidoc (hunk 3 above)
Comment (Namoshek) [reply transport_and_format.asciidoc]: I updated this again in cf5e3de822996fa550a47476d301bbfa23c21d03 for all files and occurences of _push_ and _pull_. I used `Push`/`Pull` when we talk about `Push mode` or `Push direction`, but `push[(ing|es|ed)]` everywhere else.

**Comment by RobertdeLeeuw-IHomer** [comment]
@Namoshek again, thanks for all this work! Are you on the OCPI Slack? Send me a message.

**Comment by Namoshek** [comment]
No, I'm not so far. But I wasn't done with the PR yet either. 😭

**Comment by RobertdeLeeuw-IHomer** [comment]
Send me an e-mail at robert.de.leeuw@ihomer.nl I can add you to Slack.

Sorry about the PR, I wanted to work on the specification and wanted to prevent merge conflicts.
Can you pull my changes before continuing?
